backend/src/repositories/wallet-repositry.ts

```diff
@@ -0,0 +1,18 @@
+import prisma from "../database/prisma-client";
+
+export const createWallet = async (userId: string) => {
+  return await prisma.wallet.create({ data: { userId } });
+};
+
+export const getWalletByUserId = async (userId: string) => {
+  return await prisma.wallet.findUnique({ where: { userId } });
```

Comment:
when a function a return value that is promise type no need to add await when returning it. just return it and works fine.
`Consider I'm mentioning having output. When there is no output you SHOULD await in the function`